certifikit/src/main/kotlin/app/cash/certifikit/certificates.kt

```diff
@@ -30,35 +30,32 @@ data class Certificate(
   val signatureAlgorithm: AlgorithmIdentifier,
   val signatureValue: BitString
 ) {
-  val commonName: Any?
+  val commonName: String?
     get() {
       return tbsCertificate.subject
           .flatten()
           .firstOrNull { it.type == ObjectIdentifiers.commonName }
-          ?.value
+          ?.value as String?
     }
 
-  val organizationalUnitName: Any?
+  val organizationalUnitName: String?
     get() {
       return tbsCertificate.subject
           .flatten()
           .firstOrNull { it.type == ObjectIdentifiers.organizationalUnitName }
-          ?.value
+          ?.value as String?
     }
 
-  val subjectAlternativeNames: Extension?
-    get() {
-      return tbsCertificate.extensions.firstOrNull {
-        it.id == ObjectIdentifiers.subjectAlternativeName
-      }
-    }
+  @Suppress("UNCHECKED_CAST")
+  val subjectAlternativeNames: List<Pair<Any, Any>>?
```

Comment:
I think this might be better as its own object with lists of hostnames and IP addresses?

Reply:
Follow-up is fine